src/guides/components/events.md

```diff
@@ -0,0 +1,199 @@
+# Sending Events from the Backend to Frontend
+
+In this guide, you will learn how to facilitate event-driven communication between the backend and the frontend components of a plugin.
+
+This can be accomplished using the `DefineEvents` type utility provided by the Caido SDK.
+
+The event can then be subscribed to on the frontend using `sdk.backend.onEvent()`.
+
+## Registering an API Endpoint and Event
+
+To demonstrate, we will create a plugin that sends a request, receives the response, and emits an event containing response data to the frontend.
+
+### /packages/backend/src/index.ts
+
+``` ts
+import { RequestSpec } from "caido:utils";
+import { SDK, DefineAPI, DefineEvents } from "caido:plugin";
+
+type RequestEvent = {
+  id: string;
+  status: number;
+  url: string;
+};
+
+export type BackendEvents = DefineEvents<{
+  "request-completed": (data: RequestEvent) => void;
+}>;
+
+const sendRequest = async (sdk: SDK<{}, BackendEvents>): Promise<void> => {
+  const spec = new RequestSpec("https://example.com/");
+
+  let sentRequest = await sdk.requests.send(spec);
+
+  if (sentRequest.response) {
+    sdk.api.send("request-completed", {
+      id: sentRequest.response.getId(),
+      status: sentRequest.response.getCode(),
+      url: `${spec.getHost()}:${spec.getPort()}${spec.getPath()}${spec.getQuery()}`
+    });
+  }
+}
+
+export type API = DefineAPI<{
+  sendRequest: typeof sendRequest;
+}>;
```

Comment:
We can simplify further this example by using an `sdk.events.onProxyResponse` instead of a frontend trigger. That way you will be able to remove the whole API section